.gitignore

```diff
@@ -77,3 +77,9 @@ pip-wheel-metadata/
 
 # node_modules
 node_modules
+
+# Needed for certain vim configs to work properly
+tsconfig.json
```

Comment:
nit: i am unsure if we should add this to .gitignore.
there is an option to create a user `~/.gitignore` file and add personal configuration files there.

Reply:
Thanks for raising this, I was unsure what the gitignore convention was as there are some other editor-specific ignores in there, but I'm happy to remove this of course.

Reply:
to be honest, my feeling about the .gitignore additions is based on the experience that it has not been changed that often in the past years. so the editor-specific ignores are there since a long time.